gulpfile.js

```diff
@@ -297,6 +303,216 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
     });
 });
 
+gulp.task('deploy', function(done) {
+    var argv = yargs.usage('Usage: delpoy -b [Bucket Name] -d [Upload Directory]')
+        .demand(['b', 'd']).argv;
+
+    var uploadDirectory = argv.d;
+    var bucketName = argv.b;
+
+    var iface = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+    });
+
+    if (argv.confirm) {
+        // skip prompt for travis
+        deployCesium(bucketName, uploadDirectory).then(function() {
+            done();
+        });
+    } else {
+        // prompt for confirmation
+        iface.question('Files from your computer will be published to the ' + bucketName + ' bucket. Continue? [y/n] ', function(answer) {
+            if (answer === 'y') {
+                deployCesium(bucketName, uploadDirectory).then(function() {
+                    done();
+                });
+            } else {
+                console.log('Deploy aborted by user.');
+                done();
+            }
+        });
+    }
+});
+
+// Deploy cesium to aws
+function deployCesium(bucketName, uploadDirectory) {
+    var readFile = Promise.promisify(fs.readFile);
+    var gzip = Promise.promisify(zlib.gzip);
+    var concurrencyLimit = 2000;
+    var cacheControl = "max-age=3600";
+
+    var s3 = new Promise.promisifyAll(new aws.S3({
+        maxRetries : 10,
+        retryDelayOptions : {
+            base : 500
+        }
+    }));
+
+    var existingBlobs = [];
+    var totalFiles = 0;
+    var uploaded = 0;
+    var skipped = 0;
+
+    return listAll(s3, bucketName, uploadDirectory, existingBlobs)
+    .then(function() {
+        return globby([
```

Comment:
We are uploading way more than we need to.  It would probably be better to use a specific inclusion list rather than including everything and then excluding.  Here's what the official 1.19 release uploaded looks like:

![image](https://cloud.githubusercontent.com/assets/1495196/13964225/187df6b6-f03f-11e5-9cb6-d4e2a185de1d.png)

We should just start with that.